site/_posts/2020-10-05-using-generic-connector.md

```diff
@@ -0,0 +1,339 @@
+
+## Let your users sign in
+
+The next step takes a while. Let the "The ATM" application run and let users sign in. You can, if you are following along with the code, open up an incognito browser and sign in with a few users. If you are using the example PHP application, your password is `password`. 
+
+During this time, users will be migrated over and registered with the application. Make sure that any users registering, should you allow that, are sent to register in FusionAuth rather than the legacy application.
+
+How long should this run? Long enough to move over most active users. This time frame depends on how often your users sign in, and how many of them are active. If it's an application most visit daily, users will migrate quickly. If it is an application accessed once a year, then a year may be needed to get most of your users into FusionAuth. You also want to think about how hard running both user data stores; if it's difficult, you may migrate fewer users with the phased approach.
+
+During this time period, you can find the numbers of users in FusionAuth by searching for the users with a particular attribute. Comparing this count with the legacy datastore user numbers will give you an idea of the progress of the migration. 
+
+Here's an example of a user search request which will show you the count of users with a `user.data.migratedFromTheAtmDatastore` value of true:
+
+```shell
+API_KEY=...
+
+curl -H "Authorization: $API_KEY" 'http://localhost:9011/api/user/search?queryString=data.migratedFromTheAtmDatastore%3Atrue%0A'
+```
+
+This will return all the users who have migrated. The JSON returned will look something like this
+
+```json
+{"total":629,"users":[ ... ] }
+```
+
+If you have 800 users, well, congrats, your migration is well under way. If you have 10,000 users, you've got a bit longer to wait. If you look at users in the administrative user interface, you can see users in FusionAuth registered to your application:
+
+{% include _image.liquid src="/assets/img/blogs/migrating-users-legacy-datastore/new-user-added-registration.png" alt="Success! The user is registered to your application." class="img-fluid" figure=false %}
+
+You'll also see the custom user data has been migrated over, too:
+
+{% include _image.liquid src="/assets/img/blogs/migrating-users-legacy-datastore/new-user-added-user-data.png" alt="Success! The user is has been marked as migrated. Plus they love blue." class="img-fluid" figure=false %}
+
+After this migration process has taken place for a while, or perhaps after you've have a goal percentage of users migrated, you will need to decide about users in the old datastore. If the phased migration has lasted for six months or a year and there are still users who are only in there, it is likely that these users have abandoned your application; of course only you know your user's visitation patterns, so that's just a rule of thumb. 
```

Comment:
six months or a year and there are still users ~who are only in there~ **in the old data store**